### ISSUE TITLE
How I would simulate a community

### ISSUE THREAD

**Issue by willpearse** [pearselab/r-intro-Spencerbrucehudson#6]
```{r}
sim.comm <- function(spp.lam, spp.p, spp.names, n.sites){
    #Make a matrix that you're going to output
    #Loop over all the species
        #Use your species simulation function and put that in the right column in your matrix
    #End loop
    #Do any cleanup you want on the matrix and return
}
```

**Comment by willpearse** [comment]
``` R
matrix[rows, col]
```

...can be used to subset

**Comment by Spencerbrucehudson** [comment]
Hey Dr. Pearse,

I know this isn't what you suggested, since I didn't quite understand the matrix approach, but is it suitable?

**Comment by Spencerbrucehudson** [comment]
Still not entirely comfortable with its assumption that if a species is not present at one site, then it's absent at all the others. Any recommendations for that?

**Comment by willpearse** [comment]
Your approach is fine. I am surprised you know how to use `do.call`, but I suppose this is what comes from using `plyr` et al.!

As far as your assumption, I wouldn't be either - you've made a slight slip. It's not whether a species is present at a _single_ site that determines all sites. The sequence is:
1. Pick a site
2. Is it present? (`rbinom`)
3. If it's present, what's its abundance (`rpois`)
4. Go back to 1 - start over on the next site.

...thus you go through the whole 'hurdle' process for each site. Does that make sense? I'm also online now on appear.in if you want to talk with me.

**Comment by Spencerbrucehudson** [comment]
Yeah I think so. Would a repeat loop work if I implemented it into the original function?

**Comment by Spencerbrucehudson** [comment]
Nevermind, fixed it. Is this is alright?

**Comment by willpearse** [comment]
``` r
hurdle2 <- function(sitenum, spnum){
  r <- matrix(nrow=sitenum, ncol=spnum)
  for(i in 1:spnum){
   r[,i] <- hurdle1(sitenum)
    }
  return(r)
    }
hurdle2(10, 7)
```

...right? Looks to be! Welcome to the `matrix` club. Now, isn't that easier than `dplyr`?... :p
